modules/programs/zsh/default.nix

```
@@ -391,7 +401,24 @@ in
                   - config.xdg.dataHome (XDG data directory)
                   - config.xdg.cacheHome (XDG cache directory)
                 ''
-              ];
+              ]
+            ++
+              lib.optionals
+                (
+                  config.xdg.enable
+                  && !lib.versionAtLeast config.home.stateVersion "26.05"
+                  && options.programs.zsh.dotDir.highestPrio >= 1500
```

I _think_ using `!options.programs.zsh.dotDir.isDefined` should work and be a better signal?

```suggestion
                  && !options.programs.zsh.dotDir.isDefined
```